axelrod/fingerprint.py

```diff
@@ -0,0 +1,247 @@
+        probes = probe_players.values()
+        tournament_players = [self.strategy(), dual] + list(probes)
+
+        return edges, tournament_players
+
+    def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None):
+        """Build and play the spatial tournament.
+
+        Creates the probes and their edges then builds a spatial tournament
+        where the original strategy only plays probes whose coordinates sum to
+        less than 1 (or equal). Probes whose coordinates sum to more than 1 play
+        the Dual Strategy.
+
+        Parameters
+        ----------
+        turns : integer, optional
+            The number of turns per match
+        repetitions : integer, optional
+            The number of times the round robin should be repeated
+        step : float, optional
+            The separation between each coordinate. Smaller steps will
+            produce more coordinates that will be closer together.
+        processes : integer, optional
+            The number of processes to be used for parallel processing
+        """
+        edges, tourn_players = self.construct_tournament_elements(step)
+        self.spatial_tourn = axl.SpatialTournament(tourn_players, turns=turns,
+                                                   repetitions=repetitions,
+                                                   edges=edges)
+        self.results = self.spatial_tourn.play(processes=processes,
+                                               build_results=True,
+                                               in_memory=True,
+                                               keep_interactions=True)
+        self.data = self.generate_data(self.results.interactions, self.coordinates)
+
+    def generate_data(self, interactions, coordinates):
+        """Generates useful data from a spatial tournament.
+
+        Matches interactions from `results` to their corresponding coordinate in
+        `probe_coords`.
+
+        Parameters
+        ----------
+        interactions : dictionary
+            A dictionary of the interactions of a tournament
+        probe_coords : list of tuples
+            A list of tuples of length 2, where each tuple represents a
+            coordinate, eg. (x, y).
+
+        Returns
+        ----------
+        coord_scores : dictionary
+            A dictionary where the keys are coordinates of the form (x, y) and
+            the values are the mean score for the corresponding interactions.
+        """
+        edge_scores = {key: np.mean([cfspt(i) for i in value]) for key, value in
+                       interactions.items()}
+
+        coord_scores = {coord: None for coord in coordinates}
+        for index, coord in enumerate(coordinates):
+            if sum(coord) > 1:
+                edge = (1, index + 2)
+            else:
+                edge = (0, index + 2)
+            coord_scores[coord] = edge_scores[edge]
+
+        return coord_scores
+
+    def plot(self, col_map=None):
```

Comment:
Could plot and fingerprint not be interpreted as a verb?

(I'm not overly bothered either way.)

(We also have all the plot methods in the `Plot` class which are just called by the name of the type of plot, so keeping it as `plot` or `fingerprint` would be in line with that.)